leveldb/memdb/node.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2021, Suryandaru Triandana <syndtr@gmail.com>
+// All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Package memdb provides in-memory key/value database implementation.
+package memdb
+
+import (
+	"encoding/binary"
+	"unsafe"
+)
+
+const is64Bit = int(unsafe.Sizeof(nodeInt(0))) == 8
+
+// fixedKeyLength is 8 if the nodeInt is defined as 'int' on a 64-bit system,
+// but only 4 on a 32-bit system.
+const fixedKeyLength = int(unsafe.Sizeof(nodeInt(0)))
+
+// padKey pads, if necessary, the given key to be sufficiently long
+// to work with the quickCmp function.
+func padKey(key []byte) []byte {
+	if len(key) >= fixedKeyLength {
+		return key
+	}
+	// pad
+	k := make([]byte, fixedKeyLength)
+	copy(k, key)
+	return k
```

Comment:
I don't think it should be right padding. 

Firstly, the key here is encoded in `internalKey` format. Which means the 8 bytes suffix is applied(little-endian encoded sequence number).
And secondly, let's hold the assumption the key here refers to the raw key, it should be left padding in order to keep the original key order.